test/integration/examples/apiserver_test.go

```diff
@@ -184,24 +186,38 @@ func TestAggregatedAPIServer(t *testing.T) {
 			t.Fatal(err)
 		}
 		atomic.StoreInt32(wardlePort, int32(port))
-		options := sampleserver.NewWardleServerOptions(os.Stdout, os.Stderr)
-		options.Complete()
-
-		options.RecommendedOptions.SecureServing.BindAddress = net.ParseIP("127.0.0.1")
-		options.RecommendedOptions.SecureServing.BindPort = port
-		options.RecommendedOptions.SecureServing.Listener = listener
-		options.RecommendedOptions.Etcd.StorageConfig.ServerList = []string{framework.GetEtcdURL()}
-		options.RecommendedOptions.SecureServing.ServerCert.CertDirectory = wardleCertDir
-		options.RecommendedOptions.Authentication.RequestHeader.UsernameHeaders = []string{"X-Remote-User"}
-		options.RecommendedOptions.Authentication.RequestHeader.GroupHeaders = []string{"X-Remote-Group"}
-		options.RecommendedOptions.Authentication.RequestHeader.ExtraHeaderPrefixes = []string{"X-Remote-Extra-"}
-		options.RecommendedOptions.Authentication.RequestHeader.ClientCAFile = proxyCACertFile.Name()
-		options.RecommendedOptions.Authentication.RequestHeader.AllowedNames = []string{"kube-aggregator"}
-		options.RecommendedOptions.Authentication.RemoteKubeConfigFile = kubeconfigFile.Name()
-		options.RecommendedOptions.Authorization.RemoteKubeConfigFile = kubeconfigFile.Name()
-		options.RecommendedOptions.CoreAPI.CoreAPIKubeconfigPath = kubeconfigFile.Name()
-
-		if err := options.RunWardleServer(stopCh); err != nil {
+
+		wardleCmd := sampleserver.NewCommandStartWardleServer(os.Stdout, os.Stderr, stopCh)
+		o := sampleserver.NewWardleServerOptions(os.Stdout, os.Stderr)
+		o.RecommendedOptions.SecureServing.Listener = listener
+		// parse flags
+		flags := wardleCmd.Flags()
+		o.RecommendedOptions.AddFlags(flags)
+		wardleCmd.RunE = func(c *cobra.Command, args []string) error {
+			if err := o.Complete(); err != nil {
+				return err
+			}
+			if err := o.Validate(args); err != nil {
+				return err
+			}
+			err := o.RunWardleServer(stopCh)
+			return err
+		}
+		wardleCmd.SetArgs([]string{
```

Comment:
please squash with the previous commit. I guess this change addresses my comments from before about flags?

Reply:
Ok, after all comments addressed.